packages/start/server/rpc.tsx

```diff
@@ -0,0 +1,8 @@
+import { getRequestEvent } from "solid-js/web";
+import type { RpcInfo } from "./types";
+
+export function getRpcInfo(): RpcInfo | undefined {
+  const event = getRequestEvent();
+
+  return event?.context.rpcInfo;
```

Comment:
I actually thougt about this, but since getRequestEvent doesn't throw an error if no event exists, it seemed more consistent to also throw no error here. Currently if the function is used in a non-server function or outside of a request, in both cases its just returning undefined, which users easily can check with an if condition, if they need to.

Reply:
(https://github.com/ryansolid/dom-expressions/blob/0aeca28611b5e9d85b135efbfc39892672efac95/packages/dom-expressions/src/server.js#L570)

Reply:
@ryansolid I would also be curious in your opinion on this topic 🙂

Reply:
I suppose it would be safe for us to throw an error on the server... the reason I didn't was that I could picture libraries using this in situations where people didn't opt into ALS, and also places where they call it client side.. And in those cases you would need to null check and you probably wouldn't want a thrown error.

Reply:
I am on the no-throw "side" because it feels like an early assumption at this point - the advantage of not throwing is that the user also can use the `getServerFunctionMeta` to figure out if they are in a server function, which could be useful for helper libraries, eg:

```ts
const someHelper = function(event) {
  if (getServerFunctionMeta()) {
    // Do something special if this helper is called in a server function
  }
}
```

---

That said, I am open to add the throw of course, if thats what you both prefer? If we add a throw to `getServerFunctionMeta`, how about we also add a non-throwing `isServerFunction` helper?

```ts
export const isServerFunction = () => !!getRequestEvent()?.context.serverFunctionMeta
```

Reply:
Resolving this for now after a quick chat with Ryan.